src/Core/Binding/IBindingConfigProvider.cs

```diff
@@ -45,7 +45,10 @@ public interface IBindingConfigProvider
     /// For C++ it will be in a json file in a Sonar-specific format</remarks>
     public interface IBindingConfig
     {
-        IList<string> SolutionItems { get; }
+        /// <summary>
+        /// Returns a list of all of the solution-level files that should exist if the solution is correctly bound. The files might not exist on disk. List cannot be null or empty.
+        /// </summary>
```

Comment:
I'd have thought the list would be empty if the provider doesn't handle the project...

Reply:
Not the same interface, this is the class that contains the actual data (ruleset + additional files).
We should rename it, maybe SolutionBindingData?

Reply:
No, let's leave it for now. It still has a `Save` method, which you wouldn't expect on a `data` class.